python/semantic_kernel/agents/chat_history_kernel_agent.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+import sys
+from abc import ABC, abstractmethod
+from collections.abc import AsyncIterable
+
+if sys.version_info >= (3, 12):
+    from typing import override
+else:
+    from typing_extensions import override  # pragma: no cover
+
+from semantic_kernel.agents.agent_base import AgentBase
+from semantic_kernel.agents.agent_channel import AgentChannel
+from semantic_kernel.agents.chat_history_channel import ChatHistoryChannel
+from semantic_kernel.agents.chat_history_handler import ChatHistoryHandler
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+@experimental_class
+class ChatHistoryKernelAgent(AgentBase, ChatHistoryHandler, ABC):
```

Comment:
maybe we don't need this, AgentBase and ChatHistoryHandler already have the same abstract methods setup, and this just uses the ChatHistoryChannel class, so we might be able to just set that as a ClassVar on a specific agent or just set the get channel keys and create channel methods directly in ChatCompletionAgent. At the very least there is no need to respecify the invoke and invoke_stream methods since those are the same as in ChatHistoryHandler. And then we do turn ChatHistoryHandler into a runtime checkable protocol, so that that can be used in ChatHistoryChannel to check if the supplied agent has those methods implemented. 

In dotnet currently there is only one dependent class on this as well.

Reply:
Let me sync with Chris on the context for this class. As we talked about offline, ChatHistoryHandler will go away soon.

Reply:
Hi @eavanvalkenburg , in _.Net_, `ChatHistoryHandler` is an interface that was proposed by Mark during PR review.  

I believe the idea was that any Agent that operated on a local ChatHistory would implement this interface.  Currently ChatHistoryKernelAgent is the only class that implements this and (imho) is sufficient on its own.  In the current implementation, ChatHistoryChannel targets IChatHistoryHandler (vs ChatHistoryKernelAgent)...that's about the only binding.

Looking at this with fresh eyes, I don't dispute there may be some redundancy here.   I wonder if removing `IChatHistoryHandler` in favor of `ChatHistoryKernelAgent` may be the direction to move?  I had proposed to @moonbox3 that this might be easily actionable in .Net space early next week once Ben is back.

Reply:
yeah so we moved it to a Protocol (which is somewhat similar to a interface in dotnet) which basically makes sure that the agent supplied has those two methods, alternatively we could just set agent to be a ChatCompletionAgent instead of Agent, for python that is a trivial difference, so either works, it might keep things simpler in terms of imports using the protocols, but I'm fine either way, we've removed all non-value adding things and in python you can subclass anything so still leaves plenty of room for people to subclass and adapt (without interface)!